assets/src/domain/eventEditor/ui/datetimes/dateForm/useDateFormConfig.ts

```diff
@@ -26,8 +19,16 @@ const FIELD_NAMES: Array<keyof Datetime> = ['name', 'description', 'capacity', '
 const useDateFormConfig = (id: EntityId, config?: EspressoFormProps): DateFormConfig => {
 	const { startDate: start, endDate: end, ...restProps } = useDatetimeItem({ id }) || {};
 
-	const startDate = start ? parseISO(start) : PLUS_ONE_MONTH;
-	const endDate = end ? parseISO(end) : PLUS_TWO_MONTHS;
+	let startDate = PLUS_ONE_MONTH;
+	let endDate = PLUS_TWO_MONTHS;
+
+	if (start) {
+		// "start" and "end" may be from Optimistic response as Date
+		startDate = (start as any) instanceof Date ? (start as any) : parseISO(start);
+	}
+	if (end) {
+		endDate = (end as any) instanceof Date ? (end as any) : parseISO(end);
+	}
```

Comment:
looks like an utility function might be handy here

Reply:
Done